src/main/java/seedu/address/logic/parser/ImportCommandPreparer.java

```diff
@@ -0,0 +1,124 @@
+import java.util.Optional;
+import java.util.Set;
+
+public class ImportCommandPreparer {
+    ArrayList<Person> persons = new ArrayList<>();
+    
+    public ImportCommand init() throws ParseException {
+        FileChooser fileChooser = new FileChooser();
+        fileChooser.setTitle("Select .csv file");
+        fileChooser.getExtensionFilters().add(new FileChooser.ExtensionFilter("CSV", "*.csv"));
+        fileChooser.getExtensionFilters().add(new FileChooser.ExtensionFilter("TEXT", "*.txt"));
+        File file = fileChooser.showOpenDialog(new Stage());
+        
+        return parseFile(file);
+    } 
+    
+    public ImportCommand parseFile(File file) throws ParseException {
+        
+        FileReader fr = null;
+
+        boolean hasContactWithInvalidField = false;
+        boolean hasContactWithoutName = false;
+        
+        try {
+            fr = new FileReader(file);
+        } catch (FileNotFoundException fnfe) {
+            throw new ParseException("File not found");
+        }
+
+        BufferedReader br = new BufferedReader(fr);
+
+        try {
+            String line = br.readLine();
+            while (line != null) {
+                String[] attributes = line.split(",", -1);
+                
+                if(attributes[0].equalsIgnoreCase("Name") ||
+                attributes[0].equalsIgnoreCase("Name:")){ // ignore headers
+                    line = br.readLine();
+                    continue;
+                }
+
+                int numAttributes = attributes.length;
+                
+                if(hasInvalidContact(attributes, numAttributes)) {
+                    hasContactWithoutName = true;
+                    line = br.readLine();
+                    continue;
+                }
+                
+                Name name = null;
+                Optional<Phone> phone = Optional.empty();
+                Optional<Email> email = Optional.empty();
+                Optional<Address> address = Optional.empty();
+                
+                try {
+                    name = ParserUtil.parseName(attributes[0]);
+                    if (!attributes[1].matches("")) {
+                        phone = Optional.of(ParserUtil.parsePhone(attributes[1]));
+                    }
+                    if (!attributes[2].matches("")) {
+                        email = Optional.of(ParserUtil.parseEmail(attributes[2]));
+                    }
+                    if (!attributes[3].matches("")) {
+                        address = Optional.of(ParserUtil.parseAddress(attributes[3]));
+                    }
+                } catch (ParseException pe) {
+                    hasContactWithInvalidField = true;
+                    line = br.readLine();
+                    continue;
+                }
+
+                ArrayList<String> tags = new ArrayList<>();
+                //Check for tags 
+                if (numAttributes >= 5) {
+                    for (int i = 4; i < numAttributes; i++) {
```

Comment:
4 is a magic number; use something like a constant for NUMBER_OF_CONTACT_FIELDS = 4, then from NUMBER_OF_CONTACT_FIELDS to numAttributes run your counter.